src/main/java/org/dependencytrack/util/VulnerabilityPolicyUtil.java

```diff
@@ -0,0 +1,169 @@
+package org.dependencytrack.util;
+
+import alpine.common.logging.Logger;
+import alpine.model.ConfigProperty;
+import alpine.server.util.DbUtil;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import com.fasterxml.jackson.datatype.jsonorg.JsonOrgModule;
+import com.networknt.schema.JsonSchema;
+import com.networknt.schema.JsonSchemaFactory;
+import com.networknt.schema.SpecVersion;
+import com.networknt.schema.ValidationMessage;
+import org.dependencytrack.model.ConfigPropertyConstants;
+import org.dependencytrack.model.VulnerabilityPolicy;
+import org.dependencytrack.persistence.QueryManager;
+import org.dependencytrack.policy.cel.CelPolicyScriptHost;
+import org.dependencytrack.policy.cel.CelPolicyType;
+import org.projectnessie.cel.tools.ScriptCreateException;
+import org.projectnessie.cel.tools.ScriptException;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.sql.Connection;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+
+import static org.apache.commons.io.IOUtils.resourceToString;
+
+public class VulnerabilityPolicyUtil {
+    private static final Logger logger = Logger.getLogger(VulnerabilityPolicyUtil.class);
+    private static final ObjectMapper MAPPER = new ObjectMapper(new YAMLFactory()).registerModule(new JsonOrgModule());
+
+    public static void parseVulnerabilityPolicy(StringBuilder out, List<VulnerabilityPolicy> createVulnerabilityPolicyList,
+                                          List<VulnerabilityPolicy> updateVulnerabilityPolicyList, List<String> policyNames)
+            throws ScriptCreateException, IOException {
+        JsonSchema schema = enrichYamlOjMapperAndGetSchema();
+        JsonNode jsonNode = null;
+        jsonNode = MAPPER.readTree(out.toString());
+        Set<ValidationMessage> validateMsg = schema.validate(jsonNode);
+        if (validateMsg.isEmpty()) { //yaml file provided is valid and we can proceed to validate the cel policy
+            //if cel policy is invalid, ScriptCreateException will be thrown that trickles back to inform
+            validateVulnerabilityCelPolicy(MAPPER.convertValue(jsonNode.get("conditions"), ArrayList.class));
+            VulnerabilityPolicy vulnerabilityPolicy = MAPPER.convertValue(jsonNode, VulnerabilityPolicy.class);
+            if (vulnerabilityPolicy != null) {
+                if (policyNames.contains(vulnerabilityPolicy.getName())) {
+                    updateVulnerabilityPolicyList.add(vulnerabilityPolicy);
+                } else {
+                    createVulnerabilityPolicyList.add(vulnerabilityPolicy);
+                }
+            } else {
+                logger.warn("Was not able to create vulnerability policy object successfully");
```

Comment:
Same as below, I think this should throw an exception instead and have callers of this method deal with it instead.